deepforest/cascade.py

```diff
@@ -298,8 +492,10 @@ def __getitem__(self, index):
 
     def _get_n_output(self, y):
         """Return the number of output inferred from the training labels."""
-        n_output = np.unique(y).shape[0]  # classification
-        return n_output
+        if is_classifier(self):
+            n_output = np.unique(y).shape[0]  # classification
+            return n_output
+        return y.shape[0]
```

Comment:
Setting `self.n_outputs_` to `y.shape[0]` is problematic, for example, on the boston dataset, `y_train` is of the shape `(379,)`. How about add another IF statement to distinguish univariate regression from multi-variate one? Something like:

```
if len(y.shape) == 0:
    return 1  # univariate regression
else:
    return y.shape[1]  # multi-variate regression
```

Reply:
turned out this output is not getting used in downstream function regressor, set default to 1 and added a comment

for multiple output regressor, maybe we can leave this as a next step

There's some nice thread here on have using a MultiOutputRegressor wrapper
https://github.com/dmlc/xgboost/issues/2087